src/org/openstreetmap/josm/data/validation/tests/TagChecker.java

```diff
@@ -654,15 +663,21 @@ public void check(OsmPrimitive p) {
         checkMultipolygonTags(p);
         }
 
+        final Collection<TaggingPreset> matchingPresets;
+        TagMap tags = p.getKeys();
```

Comment:
I'd put the `tags` definition inside the `if` statement, much like `matchingPresets`.